docs/src/components/action/LogoWithCopyMenu.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import copy from 'clipboard-copy';
+import { Link } from '@mui/docs/Link';
+import { Portal } from '@mui/base/Portal';
+import Box from '@mui/material/Box';
+import Snackbar from '@mui/material/Snackbar';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import Slide from '@mui/material/Slide';
+import TextFieldsRoundedIcon from '@mui/icons-material/TextFieldsRounded';
+import CheckCircleRoundedIcon from '@mui/icons-material/CheckCircleRounded';
+import { RootSvgProps } from 'docs/src/icons/RootSvg';
+import SvgMuiLogomark, {
+  muiSvgLogoString,
+  muiSvgWordmarkString,
+} from 'docs/src/icons/SvgMuiLogomark';
+
+interface LogoWithCopyMenuProps {
+  logo?: React.ComponentType<RootSvgProps>;
+  logoSvgString?: string;
+  wordmarkSvgString?: string;
+  marginLeft?: boolean;
+}
+
+export default function LogoWithCopyMenu({
+  logo: LogoSvg = SvgMuiLogomark,
+  logoSvgString = muiSvgLogoString,
+  wordmarkSvgString = muiSvgWordmarkString,
+  marginLeft,
+}: LogoWithCopyMenuProps) {
+  const [contextMenu, setContextMenu] = React.useState<{
+    mouseX: number;
+    mouseY: number;
+  } | null>(null);
+
+  const handleContextMenu = (event: React.MouseEvent) => {
+    event.preventDefault();
+    setContextMenu(
+      contextMenu === null
+        ? {
+            mouseX: event.clientX + 8,
+            mouseY: event.clientY - 8,
+          }
+        : null,
+    );
+  };
+
+  const handleClose = () => {
+    setContextMenu(null);
+  };
+
+  const [copied, setCopied] = React.useState(false);
+  const handleCopy = (svgSnippet: string) => {
+    setCopied(true);
+    copy(svgSnippet).then(() => {
+      setTimeout(() => setCopied(false), 3500);
```

Comment:
What's the `3500` for?

Reply:
It's the delay for the snack bar notifying that the element got copied.

![image](https://github.com/mui/material-ui/assets/45398769/6dde1915-9a26-46a8-8e6d-9f3b62428945)

Reply:
Could it be a constant (or at least have a comment in code)?

Reply:
Don't we need the `setTimeout` for the copy action, too, though? Like, similar to https://github.com/mui/material-ui/blob/next/docs/src/components/action/NpmCopyButton.tsx